### PR TITLE
SONARJAVA-818 Raise issue depending on java version

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/OverrideAnnotationCheck.java [hunk 1]
```diff
@@ -50,12 +53,25 @@
...
       addIssue(tree, "Add the \"@Override\" annotation above this method signature");
     }
   }
 
-  public boolean isOverriding(MethodTreeImpl methodTree) {
+  private static boolean isOverriding(MethodTreeImpl methodTree) {
     return BooleanUtils.isTrue(methodTree.isOverriding());
   }
+
+  private boolean isExcluded(JavaVersion javaVersion, MethodTreeImpl methodTree) {
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/OverrideAnnotationCheck.java]: MAJOR Make "isExcluded" a "static" method. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2325)

File: java-checks/src/main/java/org/sonar/java/checks/OverrideAnnotationCheck.java [hunk 2]
```diff
@@ -50,12 +53,25 @@
...
+    if(javaVersion.isNotSet()) {
+      return false;
+    }
+    int javaIntVersion = javaVersion.asInt();
+    return javaIntVersion <= 4 || (javaIntVersion == 5 && (methodTree.symbol().owner().isInterface() || overrideFromInterface(methodTree))) ;
+  }
+
+  private static boolean overrideFromInterface(MethodTree methodTree) {
+    return ((JavaSymbol.MethodJavaSymbol) methodTree.symbol()).overriddenSymbol().owner().isInterface();
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/OverrideAnnotationCheck.java]: BLOCKER NullPointerException might be thrown as 'overriddenSymbol' is nullable here [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* BLOCKER 1 blocker
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* BLOCKER 1 blocker

Watch the comments in this conversation to review them.

File: java-checks/src/test/java/org/sonar/java/checks/OverrideAnnotationCheckTest.java [hunk 3]
```diff
@@ -24,8 +24,25 @@
 
 public class OverrideAnnotationCheckTest {
 
+
   @Test
-  public void test() {
+  public void test_java() {
```
Comment (m-g-sonar) [inline java-checks/src/test/java/org/sonar/java/checks/OverrideAnnotationCheckTest.java]: I would rename it as ``test_unknown_java_version``

File: java-checks/src/main/java/org/sonar/java/checks/OverrideAnnotationCheck.java (hunk 2 above)
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/OverrideAnnotationCheck.java]: FP. We know that symbol is overriding. overridenSymbol can't be ``null`` then.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* BLOCKER 1 blocker

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* BLOCKER 1 blocker

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* BLOCKER 1 blocker

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* BLOCKER 1 blocker

Watch the comments in this conversation to review them.

**Pull request merged**